nameko/containers.py

```diff
@@ -171,8 +170,8 @@ def __init__(self, service_cls, config, worker_ctx_cls=None):
         self.started = False
         self._worker_pool = GreenPool(size=self.max_workers)
 
-        self._active_threads = {}
-        self._protected_threads = set()
+        self._worker_threads = WeakKeyDictionary()
```

Comment:
on a separate note, from a quick look it seems we only put the container in the `worker_ctx` to get to `config` and `service_name`. might be better just putting those in. something for another time